ledger/participant-integration-api/src/main/scala/platform/apiserver/error/ErrorInterceptor.scala

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.error
+
+import com.daml.error.{BaseError, DamlContextualizedErrorLogger}
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall
+import io.grpc.{
+  ForwardingServerCallListener,
+  Metadata,
+  ServerCall,
+  ServerCallHandler,
+  ServerInterceptor,
+  Status,
+  StatusException,
+  StatusRuntimeException,
+}
+
+final class ErrorInterceptor extends ServerInterceptor {
+
+  private val logger = ContextualizedLogger.get(getClass)
+  private val emptyLoggingContext = LoggingContext.newLoggingContext(identity)
+
+  override def interceptCall[ReqT, RespT](
+      call: ServerCall[ReqT, RespT],
+      headers: Metadata,
+      next: ServerCallHandler[ReqT, RespT],
+  ): ServerCall.Listener[ReqT] = {
+
+    val forwardingCall = new SimpleForwardingServerCall[ReqT, RespT](call) {
+
+      /** Here we are trying to detect status/trailers pairs that:
+        * - originated from the server implementation (i.e. Participant services as opposed to internal to gRPC implementation) AND
+        * - did not originate from self-service errors infrastructure.
+        * TODO error codes: Do we need to inspect/sanitize errors internal to gRPC implementation?
```

Comment:
wdym by `.. internal to gRPC implementation?`

Reply:
gRPC library also uses status codes to signal its internal errors. Just wondering if we should try to catch and sanitize them, or the opposite: make sure that we do not touch them.
That being said, I can't tell for sure if this is relevant because there might no gRPC library exceptions possible on this codepath (but I don't know that).